src/toxcore/messenger/packet/nickname.rs

```diff
@@ -0,0 +1,56 @@
+/*! Nickname struct
+*/
+
+use nom::rest;
+
+use toxcore::binary_io::*;
+
+/// Maximum size in bytes of nickname string of nickname packet
+const MAX_NICKNAME_DATA_SIZE: usize = 128;
+
+/** Nickname is a struct that holds string of my nickname.
+
+This packet is used to transmit sender's nickname to a friend.
+Every time a friend become online or my nickname is changed,
+this packet is sent to the friend or to all friends of mine.
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nickname {
+    nickname: Vec<u8>,
```

Comment:
Please make a research which encoding could be used in nickname.

Reply:
Then we should check that the byte sequence is a correct UTF-8 string and cast `Vec<u8>` to String.

Reply:
done